File: tests/guard-tags/test-context-guards.py
# Test context guards in Python

# @guard:internal:read.context
# This comment should be guarded
# And this one too
# Even this one
def public_function():
    # This code should NOT be guarded
    print("I'm not guarded!")

# @guard:sensitive:none.context
"""
This docstring should be guarded
All of these lines too
Even with multiple paragraphs

And empty lines in between
"""
def sensitive_function(name):
    # But this code is NOT guarded
    return f"Hello {name}"

class MyClass:
    # @guard:private:write.context
    """
    This class method documentation is guarded
    Multiple lines of docs
    """
    def my_method(self):
        # This method body is NOT guarded
        return 42

# Test module-level docstrings
# @guard:admin:execute.context
"""
This module-level docstring
should be guarded
until we hit code
"""
some_code = "Not guarded"

# Test mixed documentation
# @guard:team:read.context
# First line of docs
#
# Empty line above but still in docs
#
# Another line
more_code = "Also not guarded"

# @guard:special:none.context
# Comment block
# More comments
# Even more

# This is a separate comment block (not guarded)
unrelated_code = True

File: tests/guard-tags/test-context-guards.js
// Test context guards - they should only apply to documentation

// @guard:internal:read.context
// This comment should be guarded
// And this one too
// Even this one
function publicFunction() {
    // This code should NOT be guarded
    console.log("I'm not guarded!");
}

// @guard:sensitive:none.context
/**
 * This JSDoc comment should be guarded
 * All of these lines too
 * @param {string} name - Even parameter docs
 * @returns {string} And return docs
 */
function sensitiveFunction(name) {
    // But this code is NOT guarded
    return `Hello ${name}`;
}

class MyClass {
    // @guard:private:write.context
    /**
     * This method documentation is guarded
     * Multiple lines of docs
     */
    myMethod() {
        // This method body is NOT guarded
        return 42;
    }
}

// Test multi-line comment blocks
// @guard:admin:execute.context
/*
 * This entire block comment
 * should be guarded
 * until we hit code
 */
const someCode = "Not guarded";

// Test empty lines in documentation
// @guard:team:read.context
// First line of docs
//
// Empty line above but still in docs
//
// Another line
const moreCode = "Also not guarded";